Add use of `|` and `|=` operators to `Table` (#14187)
astropy/astropy: docs/table/modify_table.rst

   foo     1.0    10.0
   bar     2.0    10.0

When using ``|=``, the other object does not need to be a |Table|, it can be
anything that can be used for :ref:`construct_table` with a compatible number

> of rows.

Reviewer: It would be good to have a code example for updating with a `dict`.

> It is also possible to use ``|`` to return the new |Table| rather than update
> in-place. ``|`` requires the other object to be a |Table|::

Reviewer: ```suggestion
It is also possible to use the ``|`` operator to merge multiple |Table| instances
into a new table::
```
I would also suggest writing an example of attempting to merge a `Table` with something that's not a `Table`. If a user triggers this error and tries to find documentation about it then an explicit example with the correct error message might make it easier to find with a search engine.

  >>> t1 = Table({'name': ['foo', 'bar'], 'val': [0., 0.]}, meta={'n': 2})

>   >>> t2 = Table({'val': [1., 2.], 'val2': [10., 10.]}, meta={'id': 0})

Reviewer: I'd recommend making `t2` a `QTable` and demonstrating merging both `t1 | t2` and `t2 | t1`. Such examples would document that the output has the same type as the leftmost table to be merged.
```suggestion
  >>> t2 = QTable({'val': [1., 2.], 'val2': [10., 10.]}, meta={'id': 0})
```

  >>> t3 = t1 | t2  # Create a new table as result of update
  >>> t3
  <Table length=2>
  name   val     val2
  str3 float64 float64
  ---- ------- -------
   foo     1.0    10.0
   bar     2.0    10.0

``|`` and ``|=`` also take care of silently :ref:`merging_metadata`::

  >>> t3.meta
  {'n': 2, 'id': 0}

The columns in the updated |Table| are going to be copies of the originals. If
you need them to be references you can use the

> :meth:`~astropy.table.Table.update` method, see :ref:`copy_versus_reference`

Reviewer: ```suggestion
:meth:`~astropy.table.Table.update` method with ``copy=False``, see :ref:`copy_versus_reference`
```

   foo     1.0    10.0
   bar     2.0    10.0

``|`` and ``|=`` also takes care of silently :ref:`merging_metadata`::

  >>> t1.meta
  {'n': 2, 'id': 0}

The ``|`` and  ``|=`` operators are built on the

> :meth:`~astropy.table.Table.update` method, but the

Reviewer: This is an implementation detail that should not be mentioned in the user documentation.
